src/effects/effectchainslot.cpp

```diff
@@ -245,19 +245,11 @@ EffectChainPointer EffectChainSlot::getEffectChain() const {
 }
 
 void EffectChainSlot::clear() {
-    // Stop listening to signals from any loaded effect
     if (m_pEffectChain) {
-        m_pEffectChain->removeFromEngine(m_pEffectRack->getEngineEffectRack(),
-                                         m_iChainSlotNumber);
-        foreach (EffectSlotPointer pSlot, m_slots) {
+        for (EffectSlotPointer pSlot: m_slots) {
             pSlot->clear();
         }
-        m_pEffectChain->disconnect(this);
-        m_pEffectChain.clear();
```

Comment:
Shouldn't we check for null where th crash happens? Now the chain is kept in memory. 
It looks like the whole clear function is useless now. 
Maybe the whole chain and chain slot concept is useless now. ... 
Should this be removed or repaired?